diff --git a/core/modules/editor/src/Entity/Editor.php b/core/modules/editor/src/Entity/Editor.php
index 25ff5f8e992..092d7bae0c2 100644
--- a/core/modules/editor/src/Entity/Editor.php
+++ b/core/modules/editor/src/Entity/Editor.php
@@ -3,8 +3,10 @@
 namespace Drupal\editor\Entity;
 
 use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Config\Action\Attribute\ActionMethod;
 use Drupal\Core\Config\Entity\ConfigEntityBase;
 use Drupal\Core\StreamWrapper\StreamWrapperInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
 use Drupal\editor\EditorInterface;
 
 /**
@@ -222,4 +224,38 @@ public static function getValidStreamWrappers(): array {
     return array_keys(\Drupal::service('stream_wrapper_manager')->getNames(StreamWrapperInterface::WRITE_VISIBLE));
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  #[ActionMethod(adminLabel: new TranslatableMarkup('Add a button plugin and settings into the Active toolbar'))]
+  public function addButtonPluginIntoActiveToolbar($button_name , $button_index = -1, $plugin_name = '', $plugin_settings = []) {
+    if (!in_array($button_name, $this->settings['toolbar']['items'])) {
+      if ($button_index == -1) {
+        $this->settings['toolbar']['items'][] = $button_name;
+      }
+      elseif ($button_index == 0) {
+        array_unshift($this->settings['toolbar']['items'], $button_name);
+      }
+      else {
+        array_splice($this->settings['toolbar']['items'], $button_index, 0, $button_name);
+      }
+
+      if ($plugin_name != '') {
+        $this->settings['plugins'][$plugin_name] = $plugin_settings;
+      }
+    }
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  #[ActionMethod(adminLabel: new TranslatableMarkup('Update plugin settings in CKEditor 5 plugin settings'))]
+  public function updatePluginSettings($plugin_name, $plugin_settings = []) {
+    if (array_key_exists($plugin_name, $this->settings['plugins'])) {
+      $this->settings['plugins'][$plugin_name] = $plugin_settings;
+    }
+    return $this;
+  }
+
 }
