diff --git a/core/modules/settings_tray/src/Block/BlockEntitySettingTrayForm.php b/core/modules/settings_tray/src/Block/BlockEntitySettingTrayForm.php
index 243e907ccd3f..59579f334089 100644
--- a/core/modules/settings_tray/src/Block/BlockEntitySettingTrayForm.php
+++ b/core/modules/settings_tray/src/Block/BlockEntitySettingTrayForm.php
@@ -168,7 +168,9 @@ protected function getRedirectUrl() {
     // because it will use <current> if 'destination' is not in the query
     // string.
     if ($this->getRequest()->query->has('destination') && $destination = $this->getRedirectDestination()->get()) {
-      return Url::fromUserInput('/' . $destination);
+      // Normalize the destination to ensure it has a single leading slash.
+      $destination = (strpos($destination, '/') === 0) ? $destination : '/' . $destination;
+      return Url::fromUserInput($destination);
     }
   }
 
