diff --git a/includes/navigation.theme b/includes/navigation.theme
index f7600f948139ea6e69341dea1296e02a1c7ad5ab..77a50689293643dc8d2036a17539331ffc43529a 100644
--- a/includes/navigation.theme
+++ b/includes/navigation.theme
@@ -20,8 +20,10 @@ function gin_preprocess_navigation(&$variables) {
   $variables['core_navigation'] = _gin_module_is_active('navigation');
   $variables['is_backend'] = TRUE;
 
-  // Attach the new drupal navigation styles.
-  $variables['#attached']['library'][] = 'gin/core_navigation';
+  if (_gin_module_is_active('navigation')) {
+    // Attach the new drupal navigation styles.
+    $variables['#attached']['library'][] = 'gin/core_navigation';
+  }
 
   if (!_gin_module_is_active('navigation')) {
     $variables['#attached']['library'][] = 'gin/core_navigation_without_topbar';
