diff --git a/dist/js/sidebar.js b/dist/js/sidebar.js
index ce3f49eeab6b7df16eb7a808b8df4fa155dc73ad..3c6a7ff18640efc52a80b804de174843e7b87dfa 100644
--- a/dist/js/sidebar.js
+++ b/dist/js/sidebar.js
@@ -30,18 +30,18 @@
         },
         showSidebar: () => {
           const chooseStorage = window.innerWidth < 1024 ? "Drupal.gin.sidebarExpanded.mobile" : storageDesktop, hideLabel = Drupal.t("Hide sidebar panel"), sidebarTrigger = document.querySelector(".meta-sidebar__trigger");
-          sidebarTrigger.querySelector("span").innerHTML = hideLabel, sidebarTrigger.setAttribute("title", hideLabel), 
-          sidebarTrigger.nextSibling && (sidebarTrigger.nextSibling.innerHTML = hideLabel), 
+          null !== sidebarTrigger && (sidebarTrigger.querySelector("span").innerHTML = hideLabel, 
+          sidebarTrigger.setAttribute("title", hideLabel), sidebarTrigger.nextSibling && (sidebarTrigger.nextSibling.innerHTML = hideLabel), 
           sidebarTrigger.setAttribute("aria-expanded", "true"), sidebarTrigger.classList.add("is-active"), 
           document.body.setAttribute("data-meta-sidebar", "open"), localStorage.setItem(chooseStorage, "true"), 
-          window.innerWidth < 1280 && (Drupal.ginCoreNavigation?.collapseToolbar(), "vertical" === toolbarVariant ? Drupal.ginToolbar.collapseToolbar() : "new" === toolbarVariant && Drupal.behaviors.ginNavigation?.collapseSidebar());
+          window.innerWidth < 1280 && (Drupal.ginCoreNavigation?.collapseToolbar(), "vertical" === toolbarVariant ? Drupal.ginToolbar.collapseToolbar() : "new" === toolbarVariant && Drupal.behaviors.ginNavigation?.collapseSidebar()));
         },
         collapseSidebar: () => {
           const chooseStorage = window.innerWidth < 1024 ? "Drupal.gin.sidebarExpanded.mobile" : storageDesktop, showLabel = Drupal.t("Show sidebar panel"), sidebarTrigger = document.querySelector(".meta-sidebar__trigger");
-          sidebarTrigger.querySelector("span").innerHTML = showLabel, sidebarTrigger.setAttribute("title", showLabel), 
-          sidebarTrigger.nextSibling && (sidebarTrigger.nextSibling.innerHTML = showLabel), 
+          null !== sidebarTrigger && (sidebarTrigger.querySelector("span").innerHTML = showLabel, 
+          sidebarTrigger.setAttribute("title", showLabel), sidebarTrigger.nextSibling && (sidebarTrigger.nextSibling.innerHTML = showLabel), 
           sidebarTrigger.setAttribute("aria-expanded", "false"), sidebarTrigger.classList.remove("is-active"), 
-          document.body.setAttribute("data-meta-sidebar", "closed"), localStorage.setItem(chooseStorage, "false");
+          document.body.setAttribute("data-meta-sidebar", "closed"), localStorage.setItem(chooseStorage, "false"));
         },
         handleResize: function() {
           let windowSize = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : window;
diff --git a/gin.theme b/gin.theme
index 49db1c8b58b379465cb9d0ba09c220c64ba41cd8..2f8060788639745bd57add8414c9d94a04c833eb 100755
--- a/gin.theme
+++ b/gin.theme
@@ -52,3 +52,13 @@ function gin_css_alter(&$css, $assets) {
     $css[$custom_css]['group'] = 201;
   }
 }
+
+/**
+ * Implements hook_library_info_alter().
+ */
+function gin_library_info_alter(&$libraries, $extension) {
+  if ($extension === 'navigation' && isset($libraries['admin-reset-styles'])) {
+    unset($libraries['admin-reset-styles']['css']['base']['css/base/admin-reset-styles.css']);
+    $libraries['admin-reset-styles']['css']['base']['/' . \Drupal::service('extension.list.theme')->getPath('gin') . '/styles/admin-reset-styles.css'] = [];
+  }
+}
diff --git a/js/sidebar.js b/js/sidebar.js
index 9fd9411c68037ec7032c5e0f6d69cba80b5935b9..5f01ec55b0608f18fe0d0b54ad49aadd1fd71871 100644
--- a/js/sidebar.js
+++ b/js/sidebar.js
@@ -87,6 +87,9 @@
       const chooseStorage = window.innerWidth < breakpoint ? storageMobile : storageDesktop;
       const hideLabel = Drupal.t('Hide sidebar panel');
       const sidebarTrigger = document.querySelector('.meta-sidebar__trigger');
+      if (sidebarTrigger === null) {
+        return;
+      }
 
       sidebarTrigger.querySelector('span').innerHTML = hideLabel;
       sidebarTrigger.setAttribute('title', hideLabel);
@@ -117,6 +120,9 @@
       const chooseStorage = window.innerWidth < breakpoint ? storageMobile : storageDesktop;
       const showLabel = Drupal.t('Show sidebar panel');
       const sidebarTrigger = document.querySelector('.meta-sidebar__trigger');
+      if (sidebarTrigger === null) {
+        return;
+      }
 
       sidebarTrigger.querySelector('span').innerHTML = showLabel;
       sidebarTrigger.setAttribute('title', showLabel);
diff --git a/styles/admin-reset-styles.css b/styles/admin-reset-styles.css
new file mode 100644
index 0000000000000000000000000000000000000000..e37cb72a60dc4245a92e6ad5d952dc255b3301a0
--- /dev/null
+++ b/styles/admin-reset-styles.css
@@ -0,0 +1,34 @@
+/*
+ * This undos the reset introduced by
+ * https://www.drupal.org/node/3084859
+ */
+
+:is(#extra-specificity-hack, [data-drupal-admin-styles]) {
+  box-sizing: border-box;
+
+  *:focus {
+    outline: var(--admin-toolbar-size-focus) solid var(--admin-toolbar-color-focus);
+  }
+
+  *:not(:where(svg, svg *)) {
+    box-sizing: border-box;
+
+    button {
+      font-family: inherit;
+    }
+
+    &::after,
+    &::before {
+      box-sizing: border-box;
+    }
+  }
+
+  .visually-hidden {
+    position: absolute !important;
+    overflow: hidden;
+    clip: rect(1px, 1px, 1px, 1px);
+    width: 1px;
+    height: 1px;
+    word-wrap: normal;
+  }
+}