diff --git a/src/GinContentFormHelper.php b/src/GinContentFormHelper.php
index 9c3116edf2ca7ec6174255f34702d8bd5d3b74e7..be75737a66afc603d5c23a6d64295f31bf99bdf4 100644
--- a/src/GinContentFormHelper.php
+++ b/src/GinContentFormHelper.php
@@ -354,7 +354,7 @@ class GinContentFormHelper implements ContainerInjectionInterface {
       strpos($form_id, '_exposed_form') !== FALSE ||
       strpos($form_id, '_preview_form') !== FALSE ||
       strpos($form_id, '_delete_form') !== FALSE ||
-      strpos($form_id, '_confirm_form') !== FALSE ||
+      strpos($form_id, '_confirm') !== FALSE ||
       strpos($form_id, 'views_ui_add_') !== FALSE ||
       strpos($form_id, 'views_ui_config_') !== FALSE ||
       strpos($form_id, 'views_ui_edit_') !== FALSE ||
