diff --git a/src/Hooks.php b/src/Hooks.php
index 5a5a3df5efa03c9f7122543d3974ed769a8365e5..3cecf32f375bc4420732b3c5f82ac1ad87718e40 100644
--- a/src/Hooks.php
+++ b/src/Hooks.php
@@ -315,10 +315,12 @@ final class Hooks {
    */
   public static function applyRecipes(array &$install_state): array {
     // Apply required recipes first, followed by any additional recipes the user
-    // has chosen (i.e., via our forms).
+    // has chosen (i.e., via our forms). If no recipes were chosen by the user
+    // (e.g., the recipe selection form was skipped), fall back to the default
+    // recipes defined in the profile.
     $recipes_to_apply = array_merge(
       $install_state['profile_info']['recipes']['required'] ?? [],
-      $install_state['parameters']['recipes'] ?? [],
+      $install_state['parameters']['recipes'] ?? $install_state['profile_info']['recipes']['default'] ?? [],
     );
 
     // If the installer ran before but failed mid-stream, don't reapply any
