diff --git a/config/install/redirect.settings.yml b/config/install/redirect.settings.yml
index 70a96888dce3ced6bfe581ac1fc5a8e3d264f05e..9fbbde06ccfefa84d164fcd1895cf5d1d1ee5da8 100644
--- a/config/install/redirect.settings.yml
+++ b/config/install/redirect.settings.yml
@@ -1,4 +1,5 @@
 auto_redirect: true
+allow_from_alias: false
 default_status_code: 301
 passthrough_querystring: true
 warning: false
diff --git a/config/schema/redirect.schema.yml b/config/schema/redirect.schema.yml
index bd9ccd5d6e5fa4c6c7888a9980c47064ad51d896..050dc6ed4dbed47aee81e1cd99e04e9236cbe753 100644
--- a/config/schema/redirect.schema.yml
+++ b/config/schema/redirect.schema.yml
@@ -7,6 +7,9 @@ redirect.settings:
     auto_redirect:
       type: boolean
       label: 'Automatically create redirects when URL aliases are changed.'
+    allow_from_alias:
+      type: boolean
+      label: 'Allow redirects from aliases.'
     default_status_code:
       type: integer
       label: 'Default redirect status'
diff --git a/redirect.services.yml b/redirect.services.yml
index 908b399ec7683cb7b74a4a67fe91dd5921779b1e..8a3585ad3bde6c09173ea5f977b3d0d58c64becd 100644
--- a/redirect.services.yml
+++ b/redirect.services.yml
@@ -17,7 +17,7 @@ services:
   Drupal\redirect\RedirectChecker: '@redirect.checker'
   redirect.request_subscriber:
     class: Drupal\redirect\EventSubscriber\RedirectRequestSubscriber
-    arguments: ['@redirect.repository', '@language_manager', '@config.factory', '@path_alias.manager', '@module_handler', '@entity_type.manager', '@redirect.checker', '@router.request_context', '@path_processor_manager']
+    arguments: ['@redirect.repository', '@language_manager', '@config.factory', '@path_alias.manager', '@module_handler', '@entity_type.manager', '@redirect.checker', '@router.request_context', '@redirect.path_processor_manager']
     tags:
       - { name: event_subscriber }
   redirect.settings_cache_tag:
@@ -34,3 +34,8 @@ services:
     class: Drupal\redirect\Routing\RouteSubscriber
     tags:
       - { name: event_subscriber }
+  redirect.path_processor_manager:
+    class: Drupal\redirect\PathProcessor\RedirectPathProcessorManager
+    arguments: ['@config.factory']
+    tags:
+      - { name: service_collector, tag: path_processor_inbound, call: addInbound }
diff --git a/src/EventSubscriber/RedirectRequestSubscriber.php b/src/EventSubscriber/RedirectRequestSubscriber.php
index 9fda4f4d11db4583506b03c71fe2a2c454cb0e8f..ad76e136b963617b1fc2ba074a775119efb28bb6 100644
--- a/src/EventSubscriber/RedirectRequestSubscriber.php
+++ b/src/EventSubscriber/RedirectRequestSubscriber.php
@@ -7,11 +7,11 @@ use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Extension\ModuleHandlerInterface;
 use Drupal\Core\Language\LanguageManagerInterface;
-use Drupal\Core\PathProcessor\InboundPathProcessorInterface;
 use Drupal\Core\Routing\TrustedRedirectResponse;
 use Drupal\Core\Url;
 use Drupal\path_alias\AliasManagerInterface;
 use Drupal\redirect\Exception\RedirectLoopException;
+use Drupal\redirect\PathProcessor\RedirectPathProcessorManagerInterface;
 use Drupal\redirect\RedirectChecker;
 use Drupal\redirect\RedirectRepository;
 use Symfony\Component\HttpFoundation\Response;
@@ -66,11 +66,11 @@ class RedirectRequestSubscriber implements EventSubscriberInterface {
   protected $context;
 
   /**
-   * A path processor manager for resolving the system path.
+   * The redirect path processor manager.
    *
-   * @var \Drupal\Core\PathProcessor\InboundPathProcessorInterface
+   * @var \Drupal\redirect\PathProcessor\RedirectPathProcessorManagerInterface
    */
-  protected $pathProcessor;
+  protected $redirectPathProcessor;
 
   /**
    * Constructs a \Drupal\redirect\EventSubscriber\RedirectRequestSubscriber.
@@ -91,10 +91,10 @@ class RedirectRequestSubscriber implements EventSubscriberInterface {
    *   The redirect checker service.
    * @param \Symfony\Component\Routing\RequestContext $context
    *   Request context.
-   * @param \Drupal\Core\PathProcessor\InboundPathProcessorInterface $path_processor
-   *   The path processor.
+   * @param \Drupal\redirect\PathProcessor\RedirectPathProcessorManagerInterface $redirect_path_processor
+   *   The redirect path processor.
    */
-  public function __construct(RedirectRepository $redirect_repository, LanguageManagerInterface $language_manager, ConfigFactoryInterface $config, AliasManagerInterface $alias_manager, ModuleHandlerInterface $module_handler, EntityTypeManagerInterface $entity_type_manager, RedirectChecker $checker, RequestContext $context, InboundPathProcessorInterface $path_processor) {
+  public function __construct(RedirectRepository $redirect_repository, LanguageManagerInterface $language_manager, ConfigFactoryInterface $config, AliasManagerInterface $alias_manager, ModuleHandlerInterface $module_handler, EntityTypeManagerInterface $entity_type_manager, RedirectChecker $checker, RequestContext $context, RedirectPathProcessorManagerInterface $redirect_path_processor) {
     $this->redirectRepository = $redirect_repository;
     $this->languageManager = $language_manager;
     $this->config = $config->get('redirect.settings');
@@ -103,7 +103,7 @@ class RedirectRequestSubscriber implements EventSubscriberInterface {
     $this->entityTypeManager = $entity_type_manager;
     $this->checker = $checker;
     $this->context = $context;
-    $this->pathProcessor = $path_processor;
+    $this->redirectPathProcessor = $redirect_path_processor;
   }
 
   /**
@@ -128,27 +128,16 @@ class RedirectRequestSubscriber implements EventSubscriberInterface {
     // Get URL info and process it to be used for hash generation.
     $request_query = $request->query->all();
 
-    if (str_starts_with($request->getPathInfo(), '/system/files/') && !$request->query->has('file')) {
-      // Private files paths are split by the inbound path processor and the
-      // relative file path is moved to the 'file' query string parameter. This
-      // is because the route system does not allow an arbitrary amount of
-      // parameters. We preserve the path as is returned by the request object.
-      // @see \Drupal\system\PathProcessor\PathProcessorFiles::processInbound()
-      $path = $request->getPathInfo();
-    }
-    else {
-      // Do the inbound processing so that for example language prefixes are
-      // removed.
-      $path = $this->pathProcessor->processInbound($request->getPathInfo(), $request);
-    }
-    $path = trim($path, '/');
+    // Get the possible paths for the current request.
+    $paths = $this->redirectPathProcessor->getRedirectRequestPaths($request);
 
+    // Store current request context.
     $this->context->fromRequest($request);
 
     $cacheable_metadata = new CacheableMetadata();
 
     try {
-      $redirect = $this->redirectRepository->findMatchingRedirect($path, $request_query, $this->languageManager->getCurrentLanguage()->getId(), $cacheable_metadata);
+      $redirect = $this->redirectRepository->findMatchingRedirectMultiple($paths, $request_query, $this->languageManager->getCurrentLanguage()->getId(), $cacheable_metadata);
     }
     catch (RedirectLoopException $e) {
       \Drupal::logger('redirect')->warning('Redirect loop identified at %path for redirect %rid', ['%path' => $e->getPath(), '%rid' => $e->getRedirectId()]);
diff --git a/src/Form/RedirectSettingsForm.php b/src/Form/RedirectSettingsForm.php
index 41da6ebdb1fdc6ea471e27259938921b2e9e65e8..6a377fd96a2cea94fe5e88ee51a3db803a6490d1 100644
--- a/src/Form/RedirectSettingsForm.php
+++ b/src/Form/RedirectSettingsForm.php
@@ -52,6 +52,14 @@ class RedirectSettingsForm extends ConfigFormBase {
       // Disable only if the Path module is missing.
       '#disabled' => !$this->moduleHandler->moduleExists('path'),
     ];
+    $form['redirect_allow_from_alias'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Allow redirects from aliases.'),
+      '#description' => $this->t(
+        'Enable this option to allow redirects from Drupal URL aliases that are associated with existing internal paths, for example, a published node with a URL alias. This is disabled by default to prevent unintended redirects, for example, redirecting an existing node, thus preventing content editors from viewing/editing the node.'),
+      '#default_value' => $config->get('allow_from_alias'),
+      '#disabled' => !$this->moduleHandler->moduleExists('path'),
+    ];
     $form['redirect_passthrough_querystring'] = [
       '#type' => 'checkbox',
       '#title' => $this->t('Retain query string through redirect.'),
diff --git a/src/PathProcessor/RedirectPathProcessorManager.php b/src/PathProcessor/RedirectPathProcessorManager.php
new file mode 100644
index 0000000000000000000000000000000000000000..d7c323d4db167b885e8c376eba09faa746ac3378
--- /dev/null
+++ b/src/PathProcessor/RedirectPathProcessorManager.php
@@ -0,0 +1,111 @@
+<?php
+
+namespace Drupal\redirect\PathProcessor;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\PathProcessor\InboundPathProcessorInterface;
+use Drupal\Core\PathProcessor\PathProcessorManager;
+use Drupal\path_alias\PathProcessor\AliasPathProcessor;
+use Drupal\system\PathProcessor\PathProcessorFiles;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Redirect path processor manager.
+ *
+ * Extends PathProcessorManager to customize the processInbound logic.
+ */
+class RedirectPathProcessorManager extends PathProcessorManager implements RedirectPathProcessorManagerInterface {
+
+  /**
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+  /**
+   * Constructs a \Drupal\redirect\PathProcessor\RedirectPathProcessorManager object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->config = $config_factory->get('redirect.settings');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addInbound(InboundPathProcessorInterface $processor, $priority = 0) {
+    if ($this->applies($processor)) {
+      parent::addInbound($processor, $priority);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRedirectRequestPaths(Request $request) {
+    $paths = [];
+
+    // Do the inbound processing to get the source path.
+    $path = $request->getPathInfo();
+    $source_path = $this->processRedirectInbound($path, $request);
+    $paths[] = trim($source_path, '/');
+
+    // Add the aliased path.
+    $allow_alias = $this->config->get('allow_from_alias') ?? 0;
+    if ($allow_alias) {
+      $alias_path = $this->processRedirectInbound($path, $request, TRUE);
+      $paths[] = trim($alias_path, '/');
+    }
+
+    return array_filter(array_unique($paths));
+  }
+
+  /**
+   * Processes the inbound path.
+   *
+   * @param string $path
+   *   The path to process, with a leading slash.
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request object.
+   * @param bool $skip_alias
+   *   Whether to skip AliasPathProcessor::processInbound.
+   *
+   * @return string
+   *   The processed path.
+   */
+  protected function processRedirectInbound($path, Request $request, bool $skip_alias = FALSE) {
+    $processors = $this->getInbound();
+    foreach ($processors as $processor) {
+      // Skip AliasPathProcessor::processInbound if specified.
+      if ($skip_alias === TRUE && $processor instanceof AliasPathProcessor) {
+        continue;
+      }
+
+      $path = $processor->processInbound($path, $request);
+    }
+    return $path;
+  }
+
+  /**
+   * Returns whether this processor should be added or not.
+   *
+   * @param \Drupal\Core\PathProcessor\InboundPathProcessorInterface $processor
+   *   The processor object to add.
+   *
+   * @return bool
+   *   True if the processor should be added, false otherwise.
+   */
+  protected function applies(InboundPathProcessorInterface $processor) {
+    // Skip PathProcessorFiles::processInbound().
+    // Private files paths are split by the inbound path processor and the
+    // relative file path is moved to the 'file' query string parameter. This
+    // is because the route system does not allow an arbitrary amount of
+    // parameters.
+    if ($processor instanceof PathProcessorFiles) {
+      return FALSE;
+    }
+    return TRUE;
+  }
+
+}
diff --git a/src/PathProcessor/RedirectPathProcessorManagerInterface.php b/src/PathProcessor/RedirectPathProcessorManagerInterface.php
new file mode 100644
index 0000000000000000000000000000000000000000..7b761565fc908691f57d9b9a678f71ccad6327e0
--- /dev/null
+++ b/src/PathProcessor/RedirectPathProcessorManagerInterface.php
@@ -0,0 +1,23 @@
+<?php
+
+namespace Drupal\redirect\PathProcessor;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Defines an interface for redirect path processor manager classes.
+ */
+interface RedirectPathProcessorManagerInterface {
+
+  /**
+   * Get the paths that will be used to find matching redirects.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request object.
+   *
+   * @return array
+   *   An array of paths.
+   */
+  public function getRedirectRequestPaths(Request $request);
+
+}
diff --git a/src/RedirectRepository.php b/src/RedirectRepository.php
index ea438569eced45ad6297878dc75a25edf582ff49..590a121f2a24ffb9c1739072123b12bcee7f19ec 100644
--- a/src/RedirectRepository.php
+++ b/src/RedirectRepository.php
@@ -80,6 +80,32 @@ class RedirectRepository {
     $this->prefixList = $prefix_list;
   }
 
+  /**
+   * Finds a redirect from the given paths, query and language.
+   *
+   * @param array $source_paths
+   *   An array of redirect source paths.
+   * @param array $query
+   *   The redirect source path query.
+   * @param string $language
+   *   The language for which is the redirect.
+   * @param \Drupal\Core\Cache\CacheableMetadata|null $cacheable_metadata
+   *   The cacheable metadata for all the redirects involved.
+   *
+   * @return \Drupal\redirect\Entity\Redirect|null
+   *   The matched redirect entity or null if not found.
+   *
+   * @throws \Drupal\redirect\Exception\RedirectLoopException
+   */
+  public function findMatchingRedirectMultiple(array $source_paths, array $query = [], $language = Language::LANGCODE_NOT_SPECIFIED, ?CacheableMetadata $cacheable_metadata = NULL) {
+    $hashes = [];
+    foreach ($source_paths as $source_path) {
+      $hashes = array_merge($hashes, $this->getHashesByPath($source_path, $query, $language));
+    }
+
+    return $this->findRedirectByHashes($hashes, reset($source_paths), $language, $cacheable_metadata);
+  }
+
   /**
    * Gets a redirect for given path, query and language.
    *
@@ -98,6 +124,24 @@ class RedirectRepository {
    * @throws \Drupal\redirect\Exception\RedirectLoopException
    */
   public function findMatchingRedirect($source_path, array $query = [], $language = Language::LANGCODE_NOT_SPECIFIED, ?CacheableMetadata $cacheable_metadata = NULL) {
+    $hashes = $this->getHashesByPath($source_path, $query, $language);
+    return $this->findRedirectByHashes($hashes, $source_path, $language, $cacheable_metadata);
+  }
+
+  /**
+   * Returns redirect hashes for the given source path.
+   *
+   * @param string $source_path
+   *   The redirect source path.
+   * @param array $query
+   *   The redirect source path query.
+   * @param string $language
+   *   The language for which is the redirect.
+   *
+   * @return array
+   *   An array of redirect hashes.
+   */
+  protected function getHashesByPath($source_path, array $query, $language) {
     $source_path = ltrim($source_path, '/');
 
     // If the source path has at least two path components, check if the
@@ -105,7 +149,7 @@ class RedirectRepository {
     // that don't have any redirects on common prefixes such as /node or
     // /admin.
     if ($this->prefixList && !$this->prefixList->hasRedirectsWithPrefix($source_path)) {
-      return NULL;
+      return [];
     }
 
     $hashes = [Redirect::generateHash($source_path, $query, $language)];
@@ -113,7 +157,7 @@ class RedirectRepository {
       $hashes[] = Redirect::generateHash($source_path, $query, Language::LANGCODE_NOT_SPECIFIED);
     }
 
-    // Add a hash without the query string if using passthrough_querystring.
+    // Add a hash without the query string if using passthrough query strings.
     if (!empty($query) && $this->config->get('passthrough_querystring')) {
       $hashes[] = Redirect::generateHash($source_path, [], $language);
       if ($language != Language::LANGCODE_NOT_SPECIFIED) {
@@ -121,15 +165,30 @@ class RedirectRepository {
       }
     }
 
-    // Load redirects by hash. A direct query is used to improve performance.
-    try {
-      $rid = $this->connection->query('SELECT rid FROM {redirect} WHERE hash IN (:hashes[]) AND enabled = 1 ORDER BY LENGTH(redirect_source__query) DESC', [':hashes[]' => $hashes])->fetchField();
-    }
-    catch (\Exception) {
-      // Return early in case the query failed. This can only happen if the
-      // database update that adds the 'enabled' field hasn't run yet.
+    return $hashes;
+  }
+
+  /**
+   * Finds a redirect from the given hashes.
+   *
+   * @param array $hashes
+   *   An array of redirect hashes.
+   * @param string $source_path
+   *   The redirect source path.
+   * @param string $language
+   *   The language for which is the redirect.
+   * @param \Drupal\Core\Cache\CacheableMetadata|null $cacheable_metadata
+   *   The cacheable metadata for all the redirects involved.
+   *
+   * @return \Drupal\redirect\Entity\Redirect|null
+   *   The matched redirect entity or null if not found.
+   */
+  protected function findRedirectByHashes(array $hashes, $source_path, $language, ?CacheableMetadata $cacheable_metadata = NULL) {
+    if (empty($hashes)) {
       return NULL;
     }
+    // Load redirects by hash. A direct query is used to improve performance.
+    $rid = $this->connection->query('SELECT rid FROM {redirect} WHERE hash IN (:hashes[]) AND enabled = 1 ORDER BY LENGTH(redirect_source__query) DESC', [':hashes[]' => $hashes])->fetchField();
 
     if (!empty($rid)) {
       // Check if this is a loop.
diff --git a/tests/src/Functional/RedirectUILanguageTest.php b/tests/src/Functional/RedirectUILanguageTest.php
index 849346264b1ff5412b0683030087ebf8bafb66c8..686c1d43b5f374800ec2f67ad58f0d2cf50bb495 100644
--- a/tests/src/Functional/RedirectUILanguageTest.php
+++ b/tests/src/Functional/RedirectUILanguageTest.php
@@ -16,6 +16,11 @@ use Drupal\language\Entity\ConfigurableLanguage;
  */
 class RedirectUILanguageTest extends RedirectUITest {
 
+  /**
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
   /**
    * {@inheritdoc}
    */
@@ -31,6 +36,8 @@ class RedirectUILanguageTest extends RedirectUITest {
     $language->save();
     $language = ConfigurableLanguage::createFromLangcode('es');
     $language->save();
+
+    $this->config = $this->config('redirect.settings');
   }
 
   /**
@@ -129,4 +136,40 @@ class RedirectUILanguageTest extends RedirectUITest {
     $this->assertRedirect('de/lang-path', '/de/user', 301);
   }
 
+  /**
+   * Test language specific redirects with node aliases.
+   */
+  public function testLanguageSpecificNodeAliasRedirects(): void {
+    // Opt into allowing redirects from aliases.
+    $this->config->set('allow_from_alias', TRUE)->save();
+    $this->drupalLogin($this->adminUser);
+
+    // Create a node with alias "test-alias".
+    $this->drupalCreateNode([
+      'type' => 'article',
+      'title' => 'Test english node',
+      'path' => ['alias' => '/test-alias'],
+    ]);
+
+    // Create a redirect with en language.
+    $this->drupalGet('admin/config/search/redirect/add');
+    $this->submitForm([
+      'redirect_source[0][path]' => 'test-alias',
+      'redirect_redirect[0][uri]' => '/node',
+      'language[0][value]' => 'en',
+    ], 'Save');
+    $this->assertRedirect('test-alias', '/node', 301);
+    $this->assertRedirect('de/test-alias', NULL, 404);
+
+    // Create a redirect with de language.
+    $this->drupalGet('admin/config/search/redirect/add');
+    $this->submitForm([
+      'redirect_source[0][path]' => 'test-alias',
+      'redirect_redirect[0][uri]' => '/node/add',
+      'language[0][value]' => 'de',
+    ], 'Save');
+    $this->assertRedirect('test-alias', '/node', 301);
+    $this->assertRedirect('de/test-alias', 'de/node/add', 301);
+  }
+
 }
diff --git a/tests/src/Functional/RedirectUITest.php b/tests/src/Functional/RedirectUITest.php
index 08f80c780bdaa00847ca0da2b1aeaed314af670c..b0af83285bd26f6072252ea1af8a0e3b8f5279e0 100644
--- a/tests/src/Functional/RedirectUITest.php
+++ b/tests/src/Functional/RedirectUITest.php
@@ -27,6 +27,11 @@ class RedirectUITest extends BrowserTestBase {
    */
   protected $adminUser;
 
+  /**
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
   /**
    * The redirect repository.
    *
@@ -85,6 +90,8 @@ class RedirectUITest extends BrowserTestBase {
     $this->repository = \Drupal::service('redirect.repository');
 
     $this->storage = \Drupal::entityTypeManager()->getStorage('redirect');
+
+    $this->config = $this->config('redirect.settings');
   }
 
   /**
@@ -207,6 +214,8 @@ class RedirectUITest extends BrowserTestBase {
   public function testRedirectLoop() {
     // Redirect loop redirection only works when page caching is disabled.
     \Drupal::service('module_installer')->uninstall(['page_cache']);
+    // Opt into allowing redirects from aliases.
+    $this->config->set('allow_from_alias', TRUE)->save();
 
     /** @var \Drupal\redirect\Entity\Redirect $redirect1 */
     $redirect1 = $this->storage->create();
@@ -291,4 +300,65 @@ class RedirectUITest extends BrowserTestBase {
     $this->drupalLogin($this->adminUser);
   }
 
+  /**
+   * Test redirects with node aliases.
+   */
+  public function testNodeAliasRedirects(): void {
+    $this->drupalLogin($this->adminUser);
+    // Create a node with alias "test-alias".
+    $node = $this->drupalCreateNode([
+      'type' => 'article',
+      'title' => 'Test node redirect',
+      'langcode' => Language::LANGCODE_NOT_SPECIFIED,
+      'path' => ['alias' => '/test-alias'],
+    ]);
+
+    // Create a redirect from "test-alias" to "/node" and assert redirect.
+    $this->drupalGet('admin/config/search/redirect/add');
+    $this->submitForm([
+      'redirect_source[0][path]' => 'test-alias',
+      'redirect_redirect[0][uri]' => '/node',
+    ], 'Save');
+    // By default, aliases associated with internal routes do NOT redirect.
+    $this->assertRedirect('test-alias', NULL, 200);
+    // Opt into allowing redirects from aliases.
+    $this->config->set('allow_from_alias', TRUE)->save();
+    $this->assertRedirect('test-alias', '/node', 301);
+
+    // Update the node alias to "test-alias-updated".
+    $this->drupalGet($node->toUrl('edit-form'));
+    $this->submitForm([
+      'path[0][alias]' => '/test-alias-updated',
+    ], 'Save');
+    // Assert the "test-alias" redirect is still present.
+    $this->assertRedirect('test-alias', '/node', 301);
+
+    // Delete the node and assert redirect is still present.
+    $node->delete();
+    $this->assertRedirect('test-alias', '/node', 301);
+  }
+
+  /**
+   * Test adding a node alias when a redirect already exists.
+   */
+  public function testNodeAliasOnExistingRedirect(): void {
+    $this->drupalLogin($this->adminUser);
+
+    $this->drupalGet('admin/config/search/redirect/add');
+    $this->submitForm([
+      'redirect_source[0][path]' => 'some-url',
+      'redirect_redirect[0][uri]' => '/node',
+    ], 'Save');
+
+    $this->assertRedirect('some-url', '/node', 301);
+
+    $this->drupalCreateNode([
+      'type' => 'article',
+      'title' => 'Test node redirect',
+      'path' => ['alias' => '/some-url'],
+    ]);
+
+    $this->assertRedirect('some-url', '/node', 301);
+  }
+
 }
diff --git a/tests/src/Unit/RedirectRequestSubscriberTest.php b/tests/src/Unit/RedirectRequestSubscriberTest.php
index 263e82e268c2443ad9edf65275ab77d9d2550e1f..fcf424a778c3bc81027c310a99c8c1f5aad1dfb2 100644
--- a/tests/src/Unit/RedirectRequestSubscriberTest.php
+++ b/tests/src/Unit/RedirectRequestSubscriberTest.php
@@ -135,11 +135,12 @@ class RedirectRequestSubscriberTest extends UnitTestCase {
 
     $context = $this->createMock('Symfony\Component\Routing\RequestContext');
 
-    $inbound_path_processor = $this->createMock('Drupal\Core\PathProcessor\InboundPathProcessorInterface');
+    $inbound_path_processor = $this->createMock('Drupal\redirect\PathProcessor\RedirectPathProcessorManagerInterface');
     $inbound_path_processor->expects($this->any())
-      ->method('processInbound')
-      ->with($request->getPathInfo(), $request)
-      ->willReturnCallback(function ($path, Request $request) {
+      ->method('getRedirectRequestPaths')
+      ->with($request)
+      ->willReturnCallback(function (Request $request) {
+        $path = $request->getPathInfo();
         if (str_starts_with($path, '/system/files/') && !$request->query->has('file')) {
           // Private files paths are split by the inbound path processor and the
           // relative file path is moved to the 'file' query string parameter.
@@ -148,7 +149,7 @@ class RedirectRequestSubscriberTest extends UnitTestCase {
           // @see \Drupal\system\PathProcessor\PathProcessorFiles::processInbound()
           $path = '/system/files';
         }
-        return $path;
+        return [trim($path, '/')];
       });
 
     $alias_manager = $this->createMock(AliasManagerInterface::class);
@@ -156,7 +157,7 @@ class RedirectRequestSubscriberTest extends UnitTestCase {
     $entity_type_manager = $this->createMock(EntityTypeManagerInterface::class);
 
     $subscriber = new RedirectRequestSubscriber(
-      $this->getRedirectRepositoryStub('findMatchingRedirect', $redirect),
+      $this->getRedirectRepositoryStub('findMatchingRedirectMultiple', $redirect),
       $this->getLanguageManagerStub(),
       $this->getConfigFactoryStub(['redirect.settings' => ['passthrough_querystring' => $retain_query]]),
       $alias_manager,
