diff --git a/src/Twig/Node/StoriesNode.php b/src/Twig/Node/StoriesNode.php
index f66c647..55421a8 100644
--- a/src/Twig/Node/StoriesNode.php
+++ b/src/Twig/Node/StoriesNode.php
@@ -6,11 +6,13 @@
 use Twig\Node\Expression\AbstractExpression;
 use Twig\Node\Node;
 use TwigStorybook\Twig\TwigExtension;
+use Twig\Attribute\YieldReady;
 
 /**
  * The StoriesNode class extends the Twig Node class, providing functionality
  * specific to story nodes in Twig templates.
  */
+#[YieldReady]
 final class StoriesNode extends Node
 {
     use NodeTrait; // Use the functionality defined in NodeTrait.
diff --git a/src/Twig/Node/StoryNode.php b/src/Twig/Node/StoryNode.php
index 948dd6c..e847a4e 100644
--- a/src/Twig/Node/StoryNode.php
+++ b/src/Twig/Node/StoryNode.php
@@ -7,6 +7,7 @@
 use Twig\Node\Expression\AbstractExpression;
 use Twig\Node\Node;
 use Twig\Node\NodeOutputInterface;
+use Twig\Attribute\YieldReady;
 
 /**
  * StoryNode Class
@@ -14,6 +15,7 @@
  * This class extends the `Node` class from the Twig library and implements the `NodeOutputInterface`.
  * It represents a single "story" node in the template tree.
  */
+#[YieldReady]
 final class StoryNode extends Node implements NodeOutputInterface
 {
     use NodeTrait;
